src/runtime/auto-prompt-manager.js

```diff
@@ -44,6 +46,28 @@ const dismissEvents = [
   AnalyticsEvent.ACTION_CONTRIBUTION_OFFERS_CLOSED,
   AnalyticsEvent.ACTION_SUBSCRIPTION_OFFERS_CLOSED,
 ];
+/** @const {!Array<!AnalyticsEvent>} */
+const completedActions = [AnalyticsEvent.ACTION_SURVEY_DATA_TRANSFER];
+
+/* eslint-disable no-unused-vars */
+/** @typedef {{TYPE_REWARDED_SURVEY: (string|undefined)}} */
+let LocalStorage;
+/* eslint-enable no-unused-vars */
+
+/**
+ * Returns storage key for a given event.
+ * @param {!AnalyticsEvent} event
+ * @return {string}
+ * @protected
+ */
+function completedActionToStorageKey_(event) {
```

Comment:
nit: can we define a map instead of function? then we won't need the allowlist `completedActions` but can just do `if (COMPLETED_ACTION_TO_STORAGE_KEY.has(event)) { ... }`

Reply:
Refactored into a `Map`, though this creates a new type check warning:
```
src/runtime/auto-prompt-manager.js:51:48: WARNING - [JSC_CONFORMANCE_VIOLATION] Violation: Map is not allowed
  51| const COMPLETED_ACTION_TO_STORAGE_KEY_MAP = new Map([
```

Note, we are already using `Set` which raises the same warning.